pkg/cmd/cli/cmd/importimage.go

```diff
@@ -320,6 +320,10 @@ func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imagea
 			}
 
 			if ok {
+				// disallow re-importing anything other than DockerImage
+				if existing.From != nil && existing.From.Kind != "DockerImage" {
+					return nil, nil, fmt.Errorf("tag %q points to existing %s %q, it cannot be re-imported", tag, existing.From.Kind, existing.From.Name)
+				}
 				// disallow changing an existing tag
 				if existing.From == nil || existing.From.Kind != "DockerImage" {
```

Comment:
It seems like the second part of the expression (beginning with `||`) is no longer needed.